src/pages/blogView/blogView.component.tsx

```diff
@@ -0,0 +1,83 @@
+import { useContext, useEffect, useState } from "react"
+
+import { useLocation, useNavigate } from "react-router-dom"
+import { Bars } from 'react-loader-spinner'
+
+import { ThemeContext } from "../../context/theme/theme.context"
+
+import { getBlogByID } from "../../utils/firebase/firebaseDB.utils"
+
+const BlogView = () => {
+     const { theme } = useContext(ThemeContext);
+     const location = useLocation()
+     const navigate = useNavigate();
+
+     const [title, setTitle] = useState<string>("")
```

Comment:
Creating individual state variables for all the fields of the blog is not the right approach. Each time a state variable is set, rendering of the component is triggered. With this approach, you are triggering the render function unnecessarily. 
A better approach here would be to create one state variable called `blog` and use dot notation to access the property (title, content and etc) of the blog.

Reply:
![Screenshot 2022-10-10 at 4 04 39 PM](https://user-images.githubusercontent.com/111420831/194852388-676cfdf6-3d46-4ef2-b6e2-cf5d75eed384.png)

I'll change it like you suggested.

However, I did check this with console logs.
I believe the component is rendered only once, with useEffect being called twice. I set all the states simultaneously, so react does a single re-render.

Reply:
Change implemented.